### PR TITLE
fix(Solid): adds toggling between keyboards

### PR DESCRIPTION
**Pull request by slhay28**
## Description

Adds toggling between keyboard while also maintaining the prior positioning of the keyboard

## Changes

- adds keyboard input email story
- changes the structure of the formats prop to be `Record<string, Array<Array<string | Partial<KeyProps>>>>`
- refactors style

## Testing

- make sure keyboard renders as expected
- toggle between keyboards by going to an email story and clicking one of the special keys (`#@!`, `shift`, `áöû`). notice the changing of the keybaord

### CODE REVIEW

File: src/packages/solid/components/Keyboard/KeyboardSimple.tsx [hunk 1]
```diff
@@ -15,61 +15,110 @@
...
+              }}
+              scroll={'none'}
+              plinko
+              selected={selectedColumnIndex()}
+              gap={
+                props.gap ??
+                props.keySpacing ??
+                styles.Container.tones[tone()]?.keySpacing ??
+                styles.Container.base.keySpacing
+              }
```
Comment (anthony9187) [inline src/packages/solid/components/Keyboard/KeyboardSimple.tsx:82]: and use the memo from below here

File: src/packages/solid/components/Keyboard/KeyboardSimple.tsx [hunk 2]
```diff
@@ -15,61 +15,110 @@
...
+              // justifyContent={props.centerKeyboard ? 'center' : 'flexStart'}
```
Comment (anthony9187) [inline src/packages/solid/components/Keyboard/KeyboardSimple.tsx:83]: can we remove this?

File: src/packages/solid/components/Keyboard/KeyboardSimple.tsx [hunk 3]
```diff
@@ -15,61 +15,110 @@
...
+    } else if (key.toggle) {
+      return () => {
+        setSelectedRowIndex(rowIdx());
+        setSelectedColumnIndex(colIdx());
+        setActiveKeyboard(key.toggle);
+        keyboardRef[key.toggle]?.setFocus();
+      };
+    } else {
+      return () => setKeySignal(typeof key === 'string' ? key : key.title ?? '');
+    }
```
Comment (anthony9187) [inline src/packages/solid/components/Keyboard/KeyboardSimple.tsx:49]: is it possible to do this without `typeof` checks? I'm aware we were using them before but it's a pattern we should avoid if we can

**Review by anthony9187 — COMMENTED** [review]
can we remove the TODO from the bottom of the style file? looks like whatever it was referencing was removed at some point

File: src/packages/solid/components/Keyboard/KeyboardSimple.tsx [hunk 4]
```diff
@@ -15,61 +15,110 @@
...
-import { type KeyboardProps } from './Keyboard.jsx';
+import { For, Show, createMemo, createSignal, type Accessor, type Component } from 'solid-js';
 import Column from '../Column/Column.jsx';
 import Row from '../Row/Row.jsx';
-import Key, { type KeyProps } from '../Key/Key.jsx';
+import Key from '../Key/Key.jsx';
+import type { KeyProps } from '../Key/Key.types.js';
 import styles from './Keyboard.styles.js';
+import type { KeyboardProps } from './Keyboard.types.js';
+import { ElementNode, View } from '@lightningjs/solid';
```
Comment (anthony9187) [inline src/packages/solid/components/Keyboard/KeyboardSimple.tsx:25]: this needs to be updated to `@lightningtv/solid`

File: src/packages/solid/components/Keyboard/KeyboardSimple.tsx (hunk 3 above)
Comment (slhay28) [reply src/packages/solid/components/Keyboard/KeyboardSimple.tsx:58]: I cant think of a better way to check if its a string vs an object. I can keep looking but passing an object into the signal will break

**Comment by anthony9187** [comment]
closing this since we're archiving the repo, will add it to the new one

**Pull request closed without merging**